src/main/java/commands/CommandDone.java

```diff
@@ -0,0 +1,43 @@
+package commands;
+
+import exceptions.DukeException;
+import processor.DukeProcessor;
+import processor.Storage;
+import tasks.Task;
+
+/**
+ * Command that handles marking a task as done in the TaskList.
+ */
+public class CommandDone implements Command {
+
+    /**
+     * Attempts to find the task selected by the user, then marks it as done, saving thereafter.
+     *
+     * @param processor The instantiated DukeProcessor object.
+     * @param args      The arguments as entered by the user.
+     * @throws DukeException Throws exception if Duke is unable to find a task at the index specified.
+     */
+    public String execute(DukeProcessor processor, String args) throws DukeException {
+        String[] argsArray = args.split(" ", 2);
+        if (argsArray.length < 2) {
+            throw new DukeException("Your 'done' command is incorrect! Use the following format: done <number>");
+        }
+        int taskNumber = Integer.parseInt(argsArray[1]);
+        int taskIndex = taskNumber - 1;
+
+        Task selectedTask = processor.getTaskList().get(taskIndex);
```

Comment:
It doesn't seem like out-of-bounds or invalid taskNumber cases are handled. Correct me if I'm mistaken!

Reply:
Thanks for noticing! Have corrected it in my code :)